.circleci/config.yml

```diff
@@ -0,0 +1,117 @@
+          command: |
+            helmfile -e ${TARGET_NAMESPACE} lint
+  validate:
+    executor: aws-eks/python3
+    steps:
+      - init
+      - run:
+          name: helmfile dry-run
+          working_directory: ./helmfile
+          command: |
+            helmfile -e ${TARGET_NAMESPACE} apply --args --dry-run --suppress-secrets --detailed-exitcode=false
+  deploy:
+    executor: aws-eks/python3
+    steps:
+      - init
+      - run:
+          name: helmfile apply
+          working_directory: ./helmfile
+          command: |
+            helmfile -e ${TARGET_NAMESPACE} apply --suppress-secrets --detailed-exitcode=false
+  test:
+    executor: aws-eks/python3
+    steps:
+      - init
+      - setup_remote_docker:
+          version: 19.03.13
+          docker_layer_caching: true
+      - run:
+          name: helm test
+          command: |
+            helm test tx-crafter -n ${TARGET_NAMESPACE}
+      - run:
+          name: Extract report from volume
+          when: always
+          command: |
+            kubectl run -n ${TARGET_NAMESPACE} tmp --image=alpine:3.12.0  --overrides='{ "kind": "Pod", "apiVersion": "v1", "spec": { "containers": [{ "name": "alpine", "image": "alpine:3.12.0", "stdin":true, "volumeMounts": [{ "mountPath": "/report/output", "name": "output" }] }], "volumes": [{ "name":"output", "persistentVolumeClaim":{ "claimName": "tx-crafter-output-pvc" } }], "restartPolicy": "Never" } }'
+            for i in {1..10}; do kubectl cp ${TARGET_NAMESPACE}/tmp:/report/output/report.json $PWD/report.json && break; sleep 3; done
+      - store_artifacts:
+          path: report.json
+      - run: 
+          name: Cleanup helm test
+          when: always
+          command: |
+            kubectl delete pod tx-crafter-test -n ${TARGET_NAMESPACE}  --force=true
+            kubectl delete pod tmp -n ${TARGET_NAMESPACE} --force=true
+      - run: 
+          name: Prepare docker for build report
+          when: always
+          command: |
+            echo ${REGISTRY_PASSWORD} | docker login docker.cloudsmith.io --username ${REGISTRY_USERNAME} --password-stdin
+            docker pull docker.cloudsmith.io/consensys/docker-private/orchestrate-e2e-report:v0.0.2
+      - run: 
+          name: Build report
+          when: always
+          # Impossible to mount volumes with docker run (see https://circleci.com/docs/2.0/building-docker-images/)
+          # Use docker cp instead
+          command: |
+            docker run --name tmp -it -d docker.cloudsmith.io/consensys/docker-private/orchestrate-e2e-report:v0.0.2 sh
+            docker cp $PWD/report.json tmp:/usr/src/app/report.json 
+            docker exec tmp npm start 
+            docker cp tmp:/usr/src/app/report.html ./report.html 
+      - store_artifacts:
+          path: report.html
+
+orbs:
+  aws-eks: circleci/aws-eks@1.0.0
+  kubernetes: circleci/kubernetes@0.11.1
+  helm: circleci/helm@1.0.0
+
+version: 2.1
+
+workflows:
+  default:
+    jobs:
+      - lint
```

Comment:
Here all the jobs will be run in the branch for testing purpose before merging.
Once ready to be merged, we could have the following rules:

```
      - validate:
          ignore:
            - master
          requires: 
            - lint
      - deploy:
          only:
            - master
          requires: 
            - validate
      - test:
          only:
            - master
          requires: 
            - deploy
```